File: medium/49-group-anagram.py
from collections import defaultdict
from typing import List

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hashmap = defaultdict(list)
        for word in strs:
            key = "".join(sorted(word))
            hashmap[key].append(word)
        return list(hashmap.values())


print(Solution().groupAnagrams(["eat","tea","tan","ate","nat","bat"]))
print(Solution().groupAnagrams(["eat","tea","tan"]))
print(Solution().groupAnagrams(["","b",""]))

File: medium/49-group-anagram.cxx
#include <iostream>
#include <vector>
#include <unordered_map>
#include <algorithm>

class Solution {
	public:
		std::vector<std::vector<std::string>> groupAnagrams(std::vector<std::string>& strs) {
			std::unordered_map<std::string, std::vector<std::string>> map;
			for(const auto& word: strs) {
				auto key = word;
				std::sort(key.begin(), key.end());
				map[key].push_back(word);
			} 
			std::vector<std::vector<std::string>> vals;
			for(const auto& val: map) {
				vals.push_back(val.second);
			}
			return vals;
		}
};

int main() {
	
}
